### PR TITLE
Add readme files and table of contents

### PR DESCRIPTION
**Pull request by sonniesedge**
CHARLIE DID A PR!

It:
* Adds a table of contents to the main `readme.md` file. 
* Slightly reorganizes the main `readme.md` so that it's not immediately shouting at the visitor when they view it. 
* Adds `readme.md` files to each directory, with a mini table of contents.

This is meant as the first step to reorganizing the Playbook and making it easy to use via a published site. 

I've deliberately not attempted to rename the `house-style.md` files to `readme.md` (despite them being obvious candidates) as that could hurt direct external links. I think it would be a lot easier to slowly move content around as appropriate.

### CODE REVIEW

**Comment by davidpauljunior** [comment]
❤️ 👍 🚀 

The only thing that I wonder was whether it was slightly strange behaviour to have the first item in the TOC jump down the page (anchor link) but the rest linking to other pages.  I don't know what the solution to that is exactly, just a thought.

File: README.md [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
 
 This is a living document that we contribute to in a _public_ GitHub repo. Reasons for doing this in the open include (but are not limited to):
 
 1. Interacting with and learning from others. Receiving contributions from people who don't work here can help us, providing learning opportunities that we would not receive otherwise - for example, see [this contribution](https://github.com/springernature/frontend-playbook/pull/48#issuecomment-236139605) from @rowanmanning.
 
 1. Providing a showcase for our work/ethics. This is _really_ useful when hiring people (for both parties). We've had very positive feedback from interviewees - it's a great recruiting tool. It also means that people are quickly up and running when they join.
 
+See "[Changing the laws of engineering with pull requests](https://www.youtube.com/watch?v=YIpNpptGX6Q)" for an in depth explanation of how developing a playbook like this is of benefit.
+
 ### Where do I start?
```
Comment (davidpauljunior) [inline README.md]: I wonder if this should be removed now?
I presume that you 'start' at this README, and then navigate off via the TOC.

**Comment by sonniesedge** [comment]
Yeah, the jumpy jump bothered me as well. But I can't think of a solution that doesn't involve delinking the intro / shifting it above the TOC / making it a separate file.

File: README.md (hunk 1 above)
Comment (sonniesedge) [reply README.md]: Why just this line? Confused.

Comment (davidpauljunior) [reply README.md]: No not the line, I meant the `Where do I start` section.

Comment (sonniesedge) [reply README.md]: oh! that makes more sense. 😂

**Comment by davidpauljunior** [comment]
I think I'd be inclined to shift it above the TOC.
But if people disagree then I shall scurry back to my hole.

File: README.md [hunk 2]
```diff
@@ -1,19 +1,30 @@
...
+* [Writing](writing/README.md)
 
-See "[Changing the laws of engineering with pull requests](https://www.youtube.com/watch?v=YIpNpptGX6Q)" for an in depth explanation of how developing a playbook like this is of benefit.
+## What's this?
 
 This is a living document that we contribute to in a _public_ GitHub repo. Reasons for doing this in the open include (but are not limited to):
 
 1. Interacting with and learning from others. Receiving contributions from people who don't work here can help us, providing learning opportunities that we would not receive otherwise - for example, see [this contribution](https://github.com/springernature/frontend-playbook/pull/48#issuecomment-236139605) from @rowanmanning.
 
 1. Providing a showcase for our work/ethics. This is _really_ useful when hiring people (for both parties). We've had very positive feedback from interviewees - it's a great recruiting tool. It also means that people are quickly up and running when they join.
```
Comment (davidpauljunior) [inline README.md]: Oh I just noticed this is `1` and `1`

Comment (davidpauljunior) [reply README.md]: Merlin's beard!

**Comment by sonniesedge** [comment]
Right, I've moved the intro above the TOC and trimmed things a bit. I think it works better this way.

**Comment by jpw** [comment]
Did not get a chance to look at this yesterday, but thanks very much Charlie!

**Pull request merged**